pymc4/_hmc/integration.py

```diff
@@ -91,7 +84,7 @@ def _step(self, epsilon, state, out=None):
         # q_new = q + epsilon * v_new
         axpy(v_new, q_new, a=epsilon)
 
-        logp = self._logp_dlogp_func(q_new, q_new_grad)
+        logp, q_new_grad = self._logp_dlogp_func(q_new)
```

Comment:
I hope this is doing the right thing.

Reply:
i'll have to poke around this later, but it looks like we're providing the arrays to write to, rather than making numpy/theano allocate them. Which is to say, if `_logp_dlogp_func` does what it looks like it does, this line should be 

```
logp, q_new_grad[:] = self._logp_dlogp_func(q_new)
```

the pymc3 implementation could more explicitly be `self._logp_dlogp_func(q_new, out=q_new_grad)`